packages/features/bookings/Booker/components/OverlayCalendar/OverlayCalendarSettingsModal.tsx

```diff
@@ -43,17 +43,35 @@ const SkeletonLoader = () => {
 
 export function OverlayCalendarSettingsModal(props: IOverlayCalendarContinueModalProps) {
   const utils = trpc.useContext();
+  const [initalised, setInitalised] = useState(false);
+  const searchParams = useSearchParams();
   const setOverlayBusyDates = useOverlayCalendarStore((state) => state.setOverlayBusyDates);
   const { data, isLoading } = trpc.viewer.connectedCalendars.useQuery(undefined, {
-    enabled: !!props.open,
+    enabled: !!props.open || !!searchParams.get("overlayCalendar"),
   });
-  const { toggleValue, hasItem } = useLocalSet<{
+  const { toggleValue, hasItem, set } = useLocalSet<{
     credentialId: number;
     externalId: string;
   }>("toggledConnectedCalendars", []);
 
   const router = useRouter();
   const { t } = useLocale();
+
+  useEffect(() => {
+    if (data?.connectedCalendars && set.size === 0 && !initalised) {
+      data?.connectedCalendars.forEach((item) => {
+        item.calendars?.forEach((cal) => {
+          const id = { credentialId: item.credentialId, externalId: cal.externalId };
+          if (cal.primary) {
+            toggleValue(id);
+          }
+        });
+      });
+      setInitalised(true);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data, hasItem, set, initalised]);
```

Comment:
We have to use initialised here to ensure this only runs ONCE after the primary has been selected - allowing users to deselect the calendar if they wish